python/tk_multi_workfiles/actions/file_action_factory.py

```diff
@@ -36,158 +37,336 @@
 from ..file_item import FileItem
 from ..user_cache import g_user_cache
 
+
 class FileActionFactory(object):
     """
+    Builder for the file open menu. Based on the current selection, the build will enumerate a list of actions,
+    grouped actions and seperators that should be displayed in the UI.
     """
-    
-    def get_actions(self, file, work_area, file_model, workfiles_visible=True, publishes_visible=True):
+
+    def __init__(self, work_area, file_model, workfiles_visible=True, publishes_visible=True):
         """
+        Constructor.
+
+        :param work_area: WorkArea instance representing the context associate with the file.
+        :param file_model: A FileModel instance.
+        :param workfiles_visible: True if workfiles are visible. Defauls to True.
+        :param publishes_visible: True if publishes are visible. Defauls to True.
         """
-        if not file or not work_area or not file_model:
-            return []
+        self._work_area = work_area
+        self._file_model = file_model
+        self._workfiles_visible = workfiles_visible
+        self._publishes_visible = publishes_visible
 
         app = sgtk.platform.current_bundle()
 
         # determine if this file is in a different users sandbox:
-        in_other_users_sandbox = (work_area.contains_user_sandboxes 
-                                  and work_area.context.user and g_user_cache.current_user 
-                                  and work_area.context.user["id"] != g_user_cache.current_user["id"])
+        self._in_other_users_sandbox = (
+            work_area.contains_user_sandboxes
+            and work_area.context.user and g_user_cache.current_user
+            and work_area.context.user["id"] != g_user_cache.current_user["id"]
+        )
 
         # determine if this file is in a different work area:
-        user_work_area = work_area
-        change_work_area = (work_area.context != app.context)
-        if change_work_area and in_other_users_sandbox:
-            user_work_area = work_area.create_copy_for_user(g_user_cache.current_user)
-            change_work_area = (user_work_area.context != app.context)
+        self._user_work_area = work_area
+        self._change_work_area = (work_area.context != app.context)
```

Comment:
might be nice to have a couple of quick comments in here indicating what `_user_work_area` and `_change_work_area` represent.